src/Catalyst.Core.Modules.Consensus/Consensus.cs

```diff
@@ -0,0 +1,131 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Reactive.Linq;
+using Catalyst.Abstractions.Consensus;
+using Catalyst.Abstractions.Consensus.Cycle;
+using Catalyst.Abstractions.Consensus.Deltas;
+using Catalyst.Abstractions.Hashing;
+using Catalyst.Core.Lib.Util;
+using Catalyst.Core.Modules.Consensus.Cycle;
+using Serilog;
+
+namespace Catalyst.Core.Modules.Consensus
+{
+    /// <inheritdoc cref="IDisposable"/>
+    /// <inheritdoc cref="IConsensus"/>
+    public sealed class Consensus : IConsensus, IDisposable
+    {
+        private readonly IDeltaVoter _deltaVoter;
+        private readonly IDeltaElector _deltaElector;
+        private IDisposable _constructionProducingSubscription;
+        private IDisposable _campaigningProductionSubscription;
+        private IDisposable _votingProductionSubscription;
+
+        private readonly ICycleEventsProvider _cycleEventsProvider;
+        private readonly IDeltaHashProvider _deltaHashProvider;
+        private readonly IHashProvider _hashProvider;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the field '_hashProvider' and declare it as a local variable in the relevant methods.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4318201)